pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -144,6 +149,72 @@ func TestUpdateGroupMemberStatus(t *testing.T) {
 	}
 }
 
+func TestCheckNodeUpdate(t *testing.T) {
+	mockController := newMockMulticastController(t, false)
+	err := mockController.initialize(t)
+	assert.Nil(t, err)
+
+	for _, tc := range []struct {
+		oldNode     *corev1.Node
+		curNode     *corev1.Node
+		nodeUpdated bool
+	}{
+		{
+			oldNode: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "node0",
+					Annotations: map[string]string{types.NodeTransportAddressAnnotationKey: "172.16.0.1,1::1"},
+				},
+			},
+			curNode: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "node1",
+					Annotations: map[string]string{types.NodeTransportAddressAnnotationKey: "172.16.0.1,1::1"},
+				},
+			},
+			nodeUpdated: false,
+		},
+		{
+			oldNode: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "node0",
+					Annotations: map[string]string{types.NodeTransportAddressAnnotationKey: "172.16.0.1,1::1"},
+				},
+			},
+			curNode: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "node0",
+					Annotations: map[string]string{types.NodeTransportAddressAnnotationKey: "172.16.0.2,1::1"},
+				},
+			},
+			nodeUpdated: true,
+		},
+		{
```

Comment:
I think you should add a case for NodeIP change?

Reply:
already had one named "same name, different IPs"

Reply:
In the existing case, you are using annotation "NodeTransportAddressAnnotationKey" to Node transport address. I mean you should add a case that Node does not have the annotation, but changed Spec.NodeIP directly.

Reply:
updated. Add one named "same name, different internal IPs"